nautobot/dcim/views.py

```diff
@@ -4246,30 +4246,44 @@ def get_extra_context(self, request, instance):
             vc_form = forms.VirtualChassisForm(instance=instance)
             vc_form.fields["master"].queryset = members_queryset
 
-            context.update(
-                {
-                    "formset": formset,
-                    "vc_form": vc_form,
-                    "return_url": self.get_return_url(request, instance),
-                }
-            )
+            context["formset"] = formset
+            context["vc_form"] = vc_form
+            context["return_url"] = self.get_return_url(request, instance)
 
         return context
 
-    def form_save(self, form, **kwargs):
-        obj = super().form_save(form, **kwargs)
+    def get_form_class(self):
+        if self.action == "update":
+            return forms.VirtualChassisForm
+        return super().get_form_class()
```

Comment:
Feels like this should maybe be reversed - default form_class for the view should be `VirtualChassisForm` and only use `VirtualChassisCreateForm` for the specific `create` case?